src/DelayId.cc

```diff
@@ -60,7 +54,8 @@ DelayId::operator == (DelayId const &rhs) const
 
 DelayId::operator bool() const
 {
-    return pool_ || compositeId.getRaw();
+    assert(bool(pool_) == bool(compositeId));
```

Comment:
Can we move this assertion into the two-parameter constructor? If there is no way for external code to modify pool_ and compositeId after construction, and no internal code modifies them (i.e. the relevant state is maintained intact post construction), then it is best to assert once rather than on every boolean operator call.

Reply:
Ok.